srcpkgs/mattermost-desktop/template

```diff
@@ -0,0 +1,50 @@
+# Template file for 'mattermost-desktop'
+pkgname=mattermost-desktop
+version=4.5.2
+revision=1
+archs="x86_64"  # no musl due to electron
+wrksrc="mattermost-desktop-${version}"
+# electron-builder needs GNU tar
+hostmakedepends="git wget nodejs-lts-10 python libicns GraphicsMagick xz tar"
+short_desc="Team messaging app, an open source Slack alternative"
+maintainer="Artem Zhurikhin <ashpool@xecut.net>"
+license="Apache-2.0"
+homepage="https://mattermost.com/"
+checksum=17e32346592d175f091a4f12068e0cfc5cfd20ed7894f88bb5b54b2eb6eac74a
+
+# In the source archive the top-level dir is named "desktop-${version}/"
+# It does not contain the name of the package, so we need custom fetch & extract
```

Comment:
You can use `create_wrksrc=yes` for such cases. The contents of the tarball will be extracted there.